.github/workflows/coq.yaml

```diff
@@ -0,0 +1,34 @@
+name: Compile Coq proofs
+
+on:
+  push:
+    branches: [main]
+    paths:
+      - "**.v"
+      - "**coq.yaml"
+      - "**_CoqProject"
+      - "experimental/ni_coq/Makefile"
+  pull_request:
+    branches: [main]
+    paths:
+      - "**.v"
+      - "**coq.yaml"
+      - "**_CoqProject"
+      - "experimental/ni_coq/Makefile"
+
+jobs:
+  coq:
+    strategy:
+      matrix:
+        coq_version:
+          - '8.11'
+          - '8.12'
```

Comment:
Out of curiosity, why not just choose one version?

Reply:
Coq versions change pretty frequently, so I think it's useful to have a warning if the next upgrade is going to break something. The README claims 8.11 is supported, so we definitely need a check for that, but 8.12 is the latest stable version and we'll probably have to upgrade to it eventually, so a check for that version also seemed helpful.